### PR TITLE
Simple hackish solution to CVE-2014-3539.

### LINKED ISSUE #105: CVE-2014-3539 python-rope: pickle.load of remotely supplied data with no authentication
https://bugzilla.redhat.com/show_bug.cgi?id=1116485

### PR DESCRIPTION
**Pull request by mcepl**
This doesn't resolve the issue, but at least people using DOA should be
intentional about it and aware of the security risks.

Fixes #105

### CODE REVIEW

**Comment by mcepl** [comment]
@nixon, @aligrudi ... what do you think?

Actually, when running on RHEL I get these tests broken:

 - [ ] test_a_function_with_different_returns
 - [ ] test_a_function_with_different_returns2
 - [ ] test_arguments_with_keywords
 - [ ] test_class_dti
 - [ ] test_class_from_another_module_dti
 - [ ] test_classes_with_the_same_name
 - [ ] test_dict_keys_and_dynamicoi
 - [ ] test_dict_keys_and_dynamicoi2
 - [ ] test_dict_objects_and_dynamicoi
 - [ ] test_dti_and_concluded_data_invalidation
 - [ ] test_for_loops_and_dynamicoi
 - [ ] test_function_argument_dti
 - [ ] test_function_argument_dti2
 - [ ] test_ignoring_double_star_args
 - [ ] test_ignoring_star_args
 - [ ] test_instance_dti
 - [ ] test_invalidating_data_after_moving
 - [ ] test_list_objects_and_dynamicoi
 - [ ] test_method_dti
 - [ ] test_module_dti
 - [ ] test_nested_classes
 - [ ] test_simple_dti
 - [ ] test_strs_and_dynamicoi

**Comment by mcepl** [comment]
@aligrudi … I would like to see some comments on this so we can get rid of this CVE soon. I would make some fast release as well, so we can put it to distros and on PyPI.

**Comment by aligrudi** [comment]
Matěj Cepl <notifications@github.com> wrote:
> @aligrudi … I would like to see some comments on this so we can get rid of this CVE soon. I would make some fast release as well, so we can put it to distros and on PyPI.

Are the tests failing now?

	Ali

**Comment by mcepl** [comment]
> Are the tests failing now?

#106

**Comment by aligrudi** [comment]
Matěj Cepl <notifications@github.com> wrote:
> > Are the tests failing now?
> 
> #106 

Clearly transferring data from the sub-process does
not work on the failing machine.  Does the socket.bind()
method raise an exception in doi.py and, if so, does its
message shed some light on the problem?

	Ali

File: ropetest/advanced_oi_test.py [hunk 1]
```diff
@@ -20,6 +20,7 @@ def tearDown(self):
         super(DynamicOITest, self).tearDown()
 
     def test_simple_dti(self):
+        self.project.prefs['perform_doa'] = True
```
Comment (RonnyPfannschmidt) [inline ropetest/advanced_oi_test.py]: for the doi tests it seems sensible to just do this in the setup directly instead of seemingly each test

**Comment by soupytwist** [comment]
This bug is really old and needs to be revisited ASAP.

@mcepl, do you remember where this left off? I will take a look at it when I get a chance, but it doesn't entirely make sense to me now.

**Comment by mcepl** [comment]
Well, originally it came with https://bugzilla.redhat.com/show_bug.cgi?id=1116485 and this was my feeble effort to fix the problem.

**Comment by sebix** [comment]
Please merge this to fix the 3 year old CVE...

**Comment by mcepl** [comment]
> Please merge this to fix the 3 year old CVE...

I am not the maintainer of rope anymore, @soupytwist is.

**Comment by soupytwist** [comment]
Will merge ASAP, have to fix failing tests first.

**Comment by ulidtko** [comment]
So... uhhh. Still haven't merged "yet", right? :wink:

**Comment by soupytwist** [comment]
Aye. Honestly I don't remember why I didn't get anywhere with this. I know the test is still failing, will take another stab at it... if anyone else wants to take a look at it as well, that would be helpful.

**Comment by soupytwist** [comment]
Solved in #251.

**Pull request closed without merging**